src/WordPress/SubdomainChecker.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace CF\WordPress;
+
+// WordPressAPI::getDomainList could be use in hooks which doesn't 
+// rely on "GET /zones" request SubdomainChecker solves this problem
+// by changing WordPressAPI::domainName
+class SubdomainChecker
+{
+    private $originalDomain = '';
+    private $subDomain = '';
+
+    public function changeDomainName()
+    {
+        $config = new \CF\Integration\DefaultConfig('[]');
+        $logger = new \CF\Integration\DefaultLogger($config->getValue('debug'));
+        $dataStore = new \CF\WordPress\DataStore($logger);
+        $wordpressAPI = new \CF\WordPress\WordPressAPI($dataStore);
+        $wordpressIntegration = new \CF\Integration\DefaultIntegration($config, $wordpressAPI, $dataStore, $logger);
+        $clientAPIClient = new \CF\WordPress\WordPressClientAPI($wordpressIntegration);
+
+        $wpDomain = $wordpressAPI->getOriginalDomain();
+
+        if ($clientAPIClient->isSubdomain($wpDomain)) {
```

Comment:
`$clientAPIClient` doesn't have a `isSubdomain()` method.  The `WordPressClientAPI` should only contain logic around interacting with the Client V4 API.  If you did forget to commit this method lets move it to the `WordPressAPI` instead or move it into this class.